packages/pangea-node-sdk/src/types.ts

```diff
@@ -316,6 +317,55 @@ export namespace Redact {
   export interface StructuredParams extends StructuredOptions {
     data: Object;
   }
+
+  export enum RedactType {
+    MASK = "mask",
+    PARTIAL_MASKING = "partial_masking",
+    REPLACEMENT = "replacement",
+    DETECT_ONLY = "detect_only",
+    HASH = "hash",
+    FPE = "fpe",
+  }
+
+  export enum FPEAlphabet {
+    NUMERIC = "numeric",
+    ALPHANUMERICLOWER = "alphanumericlower",
+    ALPHANUMERIC = "alphanumeric",
+  }
+
+  export enum MaskingType {
+    MASK = "mask",
+    UNMASK = "unmask",
+  }
+
+  export interface PartialMasking {
+    masking_type?: MaskingType;
+    unmasked_from_left?: number;
+    unmasked_from_right?: number;
+    masked_from_left?: number;
+    masked_from_right?: number;
+    chars_to_ignore?: string[];
+    masking_char?: string[];
+  }
+
+  export interface RedactionMethodOverrides {
+    redaction_type: RedactType;
+    hash?: object;
+    fpe_alphabet?: FPEAlphabet;
+    partial_masking?: PartialMasking;
+    redaction_value?: string;
+  }
+
+  export interface UnredactRequest {
+    redacted_data: RedactedData;
```

Comment:
Generic would be nice (can be `T = object | string`?).

Reply:
That's what we have `export type RedactedData = object | string;` not sure if I follow what you mean

Reply:
That's a type alias, not a generic. The goal would be to make this not a type error:

```ts
const redactedData = { field1: "data1" };
const response = await redact.unredact({ redacted_data: redactedData });
response.result.redacted_data.field1;
// Property 'field1' does not exist on type 'object'.
```

The structure of the output unredacted data should match that of the input redacted data.